app/channels/application_cable/chat_room_channel.rb

```diff
@@ -0,0 +1,35 @@
+module ApplicationCable
+  class ChatRoomChannel < ApplicationCable::Channel
+    def subscribed
+      stream_for chat_room(params['chat_room_id'])
+    rescue StandardError
+      transmit error: I18n.t('api.errors.not_found')
+      reject
+    end
+
+    def unsubscribed
+      # Any cleanup needed when channel is unsubscribed
+    end
+
+    def send_message(data)
+      ActiveRecord::Base.transaction do
```

Comment:
Why is this wrapped in a transaction? There is no database call.

Reply:
Really, there's a database call in line `(18) message = current_user.messages.create!`. I want that if this call crash, don't send any notification